### PR TITLE
Destroy robots whose health is reduced to ≤0 by afterPlayed triggered abilities (fix #1315)

### LINKED ISSUE #1315: Accelerator card drops units health to 0 but doesn't kill them until they move
1. Play a robot with 1 health
2. Play the Accelerator card
3. Move the robot

### PR DESCRIPTION
**Pull request by AlexNisnevich**

### CODE REVIEW

File: src/common/util/game.ts [hunk 1]
```diff
@@ -513,15 +513,34 @@ export function dealDamageToObjectAtHex(state: w.GameState, amount: number, hex:
   return updateOrDeleteObjectAtHex(state, object, hex, cause);
 }
 
-export function updateOrDeleteObjectAtHex(state: w.GameState, object: w.Object, hex: w.HexId, cause: w.Cause | null = null): w.GameState {
+export function deleteAllDyingObjects(state: w.GameState): w.GameState {
+  Object.entries(allObjectsOnBoard(state)).forEach(([ hex, object ]) => {
```
Comment (jacobnisnevich) [inline src/common/util/game.ts:517]: Any reason why this couldn't be a `.reduce` that returns a state with all update/delete actions applied?

File: src/common/util/game.ts [hunk 2]
```diff
@@ -703,7 +722,9 @@ export function applyAbilities(state: w.GameState): w.GameState {
         // console.log(`Applying ability of ${obj.card.name} to ${ability.targets}`);
         ability.currentTargets = executeCmd(state, ability.targets, obj) as w.Target;
         const targets: w.Targetable[] = ability.currentTargets.entries;
-        targets.forEach(ability.apply);
+        targets.forEach((target) => {
```
Comment (jacobnisnevich) [inline src/common/util/game.ts:725]: Why was this change needed?

**Review by jacobnisnevich — APPROVED** [review]
Some nits, but LGTM

File: src/common/util/game.ts (hunk 2 above)
Comment (AlexNisnevich) [reply src/common/util/game.ts:725]: Oops, this was something I was looking into but ended up not doing, let me just revert.

**Review by jacobnisnevich — APPROVED** [review]
